### PR TITLE
Reduce the number of offset lengths in a stream frame

### LINKED ISSUE #414: Stream frame can have an offset length equal 0
There should be a note saying what is the value of an offset if the offset length is 0. The implementation must assume that  the offset is 0.

### PR DESCRIPTION
**Pull request by ianswett**
Also fixes #414

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -1292,10 +1292,10 @@ subsequent sections.
 ## STREAM Frame {#frame-stream}
 
 STREAM frames implicitly create a stream and carry stream data. The type byte
-for a STREAM frame contains embedded flags, and is formatted as `1FDOOOSS`.
+for a STREAM frame contains embedded flags, and is formatted as `11FDOOSS`.
 These bits are parsed as follows:
 
-* The leftmost bit must be set to 1, indicating that this is a STREAM frame.
+* The first two bit must be set to 11, indicating that this is a STREAM frame.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md]: bits, not bit

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -1407,15 +1408,13 @@ entropy on demand, which should be adequate protection against most
 opportunistic acknowledgement attacks.
 
 The type byte for a ACK frame contains embedded flags, and is formatted as
-`01NULLMM`.  These bits are parsed as follows:
+`101NLLMM`.  These bits are parsed as follows:
 
-* The first two bits must be set to 01 indicating that this is an ACK frame.
+* The first two bits must be set to 101 indicating that this is an ACK frame.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md]: three bits

**Comment by martinthomson** [comment]
I was a little surprised to see this change was merged.

@ianswett, this change to the transport doc wasn't reviewed by either @janaiyengar or myself; nor do I see any discussion on the list.  In this case, I think that it's a big improvement, so I've no objection, but I don't think we should be merging changes to documents without the approval of at least one editor.

(And sorry for not reviewing this, I don't know how I missed it.)

**Comment by ianswett** [comment]
I remember Jana and I discussing it, but next time I'll make sure one of you approves it on Github.

**Comment by martinthomson** [comment]
:) If Jana said it's OK, that's fine.  I also think that it's OK.

**Pull request merged**